modules/userdirectory-moodle/src/main/java/org/opencastproject/userdirectory/moodle/MoodleUserProviderInstance.java

```diff
@@ -142,26 +145,35 @@
    */
   private AtomicLong moodleWebServiceRequests;
 
+  private final List<String> ignoredUsernames;
+
   /**
    * Constructs an Moodle user provider with the needed settings.
    *
    * @param pid             The pid of this service.
-   * @param client          The Moodle web serivce client.
+   * @param client          The Moodle web service client.
    * @param organization    The organization.
    * @param coursePattern   The pattern of a Moodle course ID.
    * @param userPattern     The pattern of a Moodle user ID.
+   * @param groupPattern    The pattern of a Moodle group ID.
    * @param groupRoles      Whether to activate groupRoles
    * @param cacheSize       The number of users to cache.
    * @param cacheExpiration The number of minutes to cache users.
+   * @param adminUserName   Name of the global admin user.
    */
   public MoodleUserProviderInstance(String pid, MoodleWebService client, Organization organization,
-          String coursePattern, String userPattern, String groupPattern, boolean groupRoles, int cacheSize, int cacheExpiration) {
+          String coursePattern, String userPattern, String groupPattern, boolean groupRoles, int cacheSize,
+          int cacheExpiration, String adminUserName) {
     this.client = client;
     this.organization = organization;
     this.groupRoles = groupRoles;
     this.coursePattern = coursePattern;
     this.userPattern = userPattern;
     this.groupPattern = groupPattern;
+    this.ignoredUsernames = Arrays.asList("", SecurityConstants.GLOBAL_ANONYMOUS_USERNAME);
+    if (StringUtils.isNoneEmpty(adminUserName)) {
+      ignoredUsernames.add(adminUserName);
```

Comment:
This throws an `UnsupportedOperationException` as `Arrays#asList` creates an immutable list.

Reply:
Good catch. Fixed.